kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/CustomResourceDefinitionContext.java

```diff
@@ -58,16 +58,14 @@ public String getVersion() {
   public String getKind() {
     return kind;
   }
-  
+
   public static CustomResourceDefinitionBuilder v1beta1CRDFromCustomResourceType(Class<? extends CustomResource> customResource) {
     try {
       final CustomResource instance = customResource.getDeclaredConstructor().newInstance();
-    
-      final String kind = instance.getKind();
+
       final String version = instance.getVersion();
-  
+
       return new CustomResourceDefinitionBuilder()
-        .withKind(kind)
```

Comment:
D'oh, that was a dumb mistake!

Then again that shows that these builder methods are error-prone: kind, group, version should probably not be settable.